From 90af0348216a5a3c2eecd660e012500d2917b279 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Sat, 10 May 2014 10:13:44 +0200
Subject: [PATCH] Related: rhbz#1071604 don't paint anim. effect icon

... unless the slide does have custom animations. This drops dep of Draw
on libanimcore, which is in module impress, so it might not be present.
If it is not present, the Pages panel is not drawn.

Change-Id: I9d7377a4fb3289e230887752bc4a105ca1c9f812
---
 sd/source/ui/slidesorter/view/SlsPageObjectPainter.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sd/source/ui/slidesorter/view/SlsPageObjectPainter.cxx b/sd/source/ui/slidesorter/view/SlsPageObjectPainter.cxx
index 0eafa0b..60c51fa 100644
--- a/sd/source/ui/slidesorter/view/SlsPageObjectPainter.cxx
+++ b/sd/source/ui/slidesorter/view/SlsPageObjectPainter.cxx
@@ -94,7 +94,8 @@ void PageObjectPainter::PaintPageObject (
         PaintPreview(rDevice, rpDescriptor);
         PaintPageNumber(rDevice, rpDescriptor);
         PaintTransitionEffect(rDevice, rpDescriptor);
-        PaintCustomAnimationEffect(rDevice, rpDescriptor);
+        if (rpDescriptor->GetPage()->hasAnimationNode())
+            PaintCustomAnimationEffect(rDevice, rpDescriptor);
         rDevice.SetAntialiasing(nSavedAntialiasingMode);
     }
 }
-- 
1.9.0

